Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -4944,6 +4947,30 @@ void WebPage::changeSelectedIndex(int32_t index)
 }
 
 #if PLATFORM(IOS_FAMILY)
+static CGImageRef createCGImageFromJPEGDataReference(const IPC::DataReference& iconData)
+{
+    RetainPtr<CFDataRef> dataRef = adoptCF(CFDataCreate(nullptr, iconData.data(), iconData.size()));
+    RetainPtr<CGDataProviderRef> imageProviderRef = adoptCF(CGDataProviderCreateWithCFData(dataRef.get()));
+
+    auto imageSrc = adoptCF(CGImageSourceCreateWithDataProvider(imageProviderRef.get(), nullptr));
+    if (!imageSrc)
+        return nullptr;
+    
+    auto type = adoptCF(CGImageSourceGetType(imageSrc.get()));
```

Comment:
Now that we understand that WebKit creates these images, and they are always under our control, we probably don't need this check.